src/components/partnership/partnership.service.ts

```diff
@@ -603,4 +627,71 @@ export class PartnershipService {
       );
     }
   }
+
+  protected async isFirstPartnership(projectId: string): Promise<boolean> {
+    const result = await this.db
+      .query()
+      .match([
+        node('project', 'Project', { id: projectId }),
+        relation('out', '', 'partnership', { active: true }),
+        node('partnership'),
+      ])
+      .return(['partnership'])
+      .asResult<{ partnership?: Node }>()
+      .first();
+
+    return result?.partnership ? false : true;
+  }
+
+  /**
+   *
+   * match current primary partnership, its project, and all other partnerships
+   * set current to primary false
+   * set first of other partnerships to primary true
+   */
+  protected async movePrimaryPartnership(
+    value: boolean,
```

Comment:
Maybe this could be changed to 
```suggestion
    direction: 'to' | 'from',
```
from that we could set the booleans on line 685 & 693 accordingly. 
I think that would be clearer as well:
```ts
// create/update
movePrimaryPartnership('to', id);

// delete
movePrimaryPartnership('from', id);
```